libsolidity/ast/ASTForward.h

```diff
@@ -99,6 +99,25 @@ class StructuredDocumentation;
 
 class VariableScope;
 
+template <class T>
```

Comment:
Just defining it for ASTNode does not work because of the implicit conversion?

Reply:
It works for everything except the mac osx build